app/storyedit/models/story.model.ts

```diff
@@ -0,0 +1,125 @@
+import {Observable} from 'rxjs';
+import {CmsService} from '../../shared/cms/cms.service';
+import {HalDoc} from '../../shared/cms/haldoc';
+
+class StoryPayload {
+  id: number = null;
+  title: string = '';
+  shortDescription: string = '';
+  modifiedAt: Date = null;
+  toJSON(): {} {
+    return {
+      title: this.title,
+      shortDescription: this.shortDescription
+    };
+  }
+}
+
+class StoryChanged {
+  title: boolean = false;
+  shortDescription: boolean = false;
+}
+
+class StoryInvalid {
+  title: string;
+  shortDescription: string;
+}
+
+export class StoryModel {
```

Comment:
Duplication in this class for the Payload, Changed, Invalid, and the `set` methods.
Seems like there should be some way to do this with less duplicate code per attribute?